### DISCUSSION TITLE
Site Meta

### DISCUSSION THREAD

**Discussion by HarryWangATX** [auroral-ui/hexo-theme-aurora#58] [Q&A]
I see that in the documentation, you can inject scripts into the site meta. Is there any way to insert `<meta>`  tags into the `<head>` to improve SEO? Maybe through MD front matter if possible?

**Comment by bennyxguo** [comment]
Which type of meta would you like to add?

> I see that in the documentation, you can inject scripts into the site meta. Is there any way to insert `<meta>` tags into the `<head>` to improve SEO? Maybe through MD front matter if possible?

**Comment by HarryWangATX** [comment]
```html
<meta name="description" content="Free Web tutorials">
<meta name="keywords" content="HTML, CSS, JavaScript">
<meta name="author" content="John Doe">
```

@TriDiamond, these ones, but user custom sets what to put.

**Reply by bennyxguo** [reply]
These types can be added as a feature to the theme, but for the injections those are fixed, can't adapt to the change of a specific post or data.

I will add these in when the post are loaded, and for injection I will add a meta part where you can add in the global meta (global meta means it's fixed, wont change base on the data)

**Reply by HarryWangATX** [reply]
yea that sounds good. I think it might be possible to set these when we generate the static files?

**Reply by HarryWangATX** [reply]
-------
description: 'lorem ipsum'
keywords:
  - sdfa
  - asdfasd
author: name
-------

maybe something like that ^ which gets integrated during the build.

**Reply by bennyxguo** [reply]
> yea that sounds good. I think it might be possible to set these when we generate the static files?

For global meta, it's possible, for post related meta, it's not possible. 

Injections are injected during build time, but the rest need to be added during runtime.

**Reply by HarryWangATX** [reply]
i see.

**Reply by HarryWangATX** [reply]
> I will add these in when the post are loaded, 

so is this the workaround for post related meta?

**Reply by bennyxguo** [reply]
> > I will add these in when the post are loaded,
> 
> so is this the workaround for post related meta?

It's not a work around, it's a feature that need to be added to achieve this result. Currently is not supported.

**Reply by HarryWangATX** [reply]
okay, got it.

**Comment by btheNerd** [comment]
@bennyxguo 
Hi, first of all, I hope that you know I am really enjoying applying aurora theme and thank you for letting us use such a beautiful theme.
I have a problem to improve my SEO experience, so I need your help.

I have tried adding :      

```
<meta name="google-site-verification" content="myverificationcode" />
<meta name="naver-site-verification" content="myverificationcode" />
```
not only to 
```
site_meta:
    cdn: en
.
.
.
    google-site-verication : 'myverificationcode'
    naver-site-verication : 'myverificationcode'
.
.
.
injects:
    scripts:
        <meta name="google-site-verification" content="myverificationcode" />
        <meta name="naver-site-verification" content="myverificationcode" />
    css:
        <meta name="google-site-verification" content="myverificationcode" />
        <meta name="naver-site-verification" content="myverificationcode" />
```
of _config.aurora.yml file, 
but also to
```
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <%- autoCanonical(config, page) %> 
  <%- partial('google-analytics') %>
  <meta name="google-site-verification" content="myverificationcode" />
  <meta name="naver-site-verification" content="myverificationcode" />

```
of head.ejs .
However, they all failed to make those contents appear.
Is there another method that I should try? 

Thanks for reading and hope you have a nice weekend!
